### PR TITLE
Add support for cleaning certificates on the machine to the dev-certs tool

### PR DESCRIPTION
**Pull request by javiercn**
Fixes https://github.com/aspnet/DotNetTools/issues/366

### CODE REVIEW

**Comment by javiercn** [comment]
The gesture is `dotnet dev-certs https --clean` as per @danroth27 request

File: src/dotnet-dev-certs/Program.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ internal class Program
         private const int ErrorUserCancelledTrustPrompt = 5;
         private const int ErrorNoValidCertificateFound = 6;
         private const int ErrorCertificateNotTrusted = 7;
+        private const int ErrorCleaningUpTheCertificates = 8;
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: ErrorCleaningUp~~The~~Certificates

File: src/dotnet-dev-certs/Program.cs [hunk 2]
```diff
@@ -95,6 +107,22 @@ public static int Main(string[] args)
...
+            var manager = new CertificateManager();
+            try
+            {
+                manager.CleanHttpsCertificates();
+                reporter.Verbose("HTTPS development certificates successfully removed from the machine.");
+                return Success;
+            }
+            catch
+            {
+                reporter.Error("There was an error trying to clean the HTTPS certificates on this machine.");
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: trying to clean ~~the~~ HTTPS development certificates

File: src/dotnet-dev-certs/Program.cs [hunk 3]
```diff
@@ -95,6 +107,22 @@ public static int Main(string[] args)
             }
         }
 
+        private static int CleanHttpsCertificates(CommandOption clean, IReporter reporter)
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: Where is `clean` coming from? You don't pass it in?

Comment (javiercn) [reply src/dotnet-dev-certs/Program.cs]: Removed

File: src/dotnet-dev-certs/Program.cs [hunk 4]
```diff
@@ -73,8 +79,14 @@ public static int Main(string[] args)
                         var reporter = new ConsoleReporter(PhysicalConsole.Singleton, verbose.HasValue(), quiet.HasValue());
                         if (check.HasValue())
                         {
-                            return CheckHttpsCertificate(check, trust, reporter);
+                            return CheckHttpsCertificate(trust, reporter);
+                        }
+
+                        if (clean.HasValue())
```
Comment (mkArtakMSFT) [inline src/dotnet-dev-certs/Program.cs]: looks like if both clean and check flags will be specified, the tool will always do the "check" only. Should we rather show some error?

File: src/dotnet-dev-certs/Program.cs [hunk 5]
```diff
@@ -95,7 +107,33 @@ public static int Main(string[] args)
...
+                {
+                    reporter.Output("Cleaning HTTPS development certificates from the machine. A prompt might get" +
+                        "displayed to confirm the removal of some of the certificates.");
+                }
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    reporter.Output("Cleaning HTTPS development certificates from the machine. This operation might" +
+                        "require elevated privileges. If that is the case, a prompt for credentials will be displayed.");
+                }
+                //manager.CleanHttpsCertificates();
```
Comment (mkArtakMSFT) [inline src/dotnet-dev-certs/Program.cs]: You don't actually clean- is the comment intentional?

File: src/dotnet-dev-certs/Program.cs [hunk 6]
```diff
@@ -50,6 +51,11 @@ public static int Main(string[] args)
                         "Check for the existence of the certificate but do not perform any action",
                         CommandOptionType.NoValue);
 
+                    var clean = c.Option(
```
Comment (natemcmaster) [inline src/dotnet-dev-certs/Program.cs]: I'd love to see this separated into its own subcommand. I think we have already opened https://github.com/aspnet/DotNetTools/issues/357 to discuss this.

File: src/dotnet-dev-certs/Program.cs (hunk 5 above)
Comment (javiercn) [reply src/dotnet-dev-certs/Program.cs]: I'm waiting for the changes I did in common to go through

File: src/dotnet-dev-certs/Program.cs (hunk 4 above)
Comment (javiercn) [reply src/dotnet-dev-certs/Program.cs]: I've added error checking above to weed out these cases

File: src/dotnet-dev-certs/Program.cs (hunk 6 above)
Comment (muratg) [reply src/dotnet-dev-certs/Program.cs]: Agreed, and I'd like to hear @danroth27's thought on this as well.

File: src/dotnet-dev-certs/Program.cs [hunk 7]
```diff
@@ -95,7 +116,34 @@ public static int Main(string[] args)
...
+            try
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    reporter.Output("Cleaning HTTPS development certificates from the machine. A prompt might get" +
+                        "displayed to confirm the removal of some of the certificates.");
+                }
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    reporter.Output("Cleaning HTTPS development certificates from the machine. This operation might" +
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: Space after `might`

> Cleaning HTTPS development certificates from the machine. A prompt may be displayed if elevated privileges are required.

File: src/dotnet-dev-certs/Program.cs [hunk 8]
```diff
@@ -95,7 +116,34 @@ public static int Main(string[] args)
...
 
-        private static int CheckHttpsCertificate(CommandOption check, CommandOption trust, IReporter reporter)
+        private static int CleanHttpsCertificates(IReporter reporter)
+        {
+            var manager = new CertificateManager();
+            try
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    reporter.Output("Cleaning HTTPS development certificates from the machine. A prompt might get" +
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: Space after `get`

Comment (pranavkm) [reply src/dotnet-dev-certs/Program.cs]: > Cleaning HTTPS development certificates from the machine. A prompt may be displayed to confirm removal of some certificates.

File: src/dotnet-dev-certs/Program.cs [hunk 9]
```diff
@@ -95,7 +116,34 @@ public static int Main(string[] args)
...
+                        "require elevated privileges. If that is the case, a prompt for credentials will be displayed.");
+                }
+
+                manager.CleanupHttpsCertificates();
+                reporter.Verbose("HTTPS development certificates successfully removed from the machine.");
+                return Success;
+            }
+            catch
+            {
+                reporter.Error("There was an error trying to clean HTTPS development certificates on this machine.");
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: > An error occured when attempting to clean HTTPS development certificates. 

Should we also print the exception message? Might be useful to diagnose \ guide users.

File: src/dotnet-dev-certs/Program.cs [hunk 10]
```diff
@@ -142,6 +191,13 @@ private static int EnsureHttpsCertificate(CommandOption exportPath, CommandOptio
                     "on the system keychain.");
             }
 
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && trust?.HasValue() == true)
+            {
+                reporter.Warn("Trusting the HTTPS development certificate was requested. If the certificate is not " +
```
Comment (pranavkm) [inline src/dotnet-dev-certs/Program.cs]: > Trusting the HTTPS development certificate was requested. A confirmation prompt will be displayed if the certificate was not previously trusted. Click yes on the prompt to trust the certificate.

File: src/dotnet-dev-certs/Program.cs (hunk 9 above)
Comment (muratg) [reply src/dotnet-dev-certs/Program.cs]: Sounds reasonable to me.

**Pull request merged**